src/server/resolvers/index.ts

```diff
@@ -256,33 +256,46 @@ export const resolvers: CustomResolvers<Context> = {
 			// no email sent if declined
 			return value;
 		},
-		createTier: async (root, { input: { name, permissions } }, { models, user }: Context) => {
-			if (!user || user.userType !== UserType.Organizer)
-				throw new AuthenticationError(`user '${JSON.stringify(user)}' must be organizer`);
-			await models.Tiers.insertOne({
+		createTier: async (
+			root,
+			{ input: { name, permissions } },
+			{ models, user }: Context
+		): Promise<TierDbObject> => {
+			checkIsAuthorized(UserType.Organizer, user);
+			const tier = await models.Tiers.findOne({ name });
+			// currently does not support updating
+			if (tier) throw new UserInputError(`Tier ${name} already exists, no action performed.`);
+
+			const newTier: TierDbObject = {
 				_id: new ObjectID(),
 				name,
 				permissions: permissions || [],
-			});
-			const tierCreated = await models.Tiers.findOne({ name });
-			if (!tierCreated) throw new AuthenticationError(`tier not found: ${name}`);
-			return tierCreated;
+			};
+			logger.info(`creating tier ${JSON.stringify(newTier)}`);
+			await models.Tiers.insertOne(newTier);
+			return newTier;
 		},
-		createCompany: async (root, { input: { name, tierId } }, { models, user }: Context) => {
-			if (!user || user.userType !== UserType.Organizer)
-				throw new AuthenticationError(`user '${JSON.stringify(user)}' must be organizer`);
-
+		createCompany: async (
+			root,
+			{ input: { name, tierId } },
+			{ models, user }: Context
+		): Promise<CompanyDbObject> => {
+			checkIsAuthorized(UserType.Organizer, user);
 			const tier = await models.Tiers.findOne({ _id: new ObjectID(tierId) });
 			if (!tier) throw new UserInputError(`Tier with id ${tierId}' doesn't exist.`);
-			await models.Companies.insertOne({
+			// currently does not support updating
```

Comment:
I feel like adding updating would be fairly easy to implement and it would make the flow for setting up the hackathon easier, so we should definitely do it at some point.